trees/GenericTrees.java

```diff
@@ -0,0 +1,81 @@
+import java.util.*;
```

Comment:
To such complex ques, add a small question prompt too

Reply:
Right!, will add from the next time.